docs/add_custom_features.md

```diff
@@ -0,0 +1,66 @@
+# Add custom features for custom analysis
+
+If you want to perform custom analysis with your custom features that are not supported in the original task processors, and these features are only related to one particular system/dataset, you can define the custom features and analysis in the `metadata` section in the output JSON file. 
+
+Here is the template of the output JSON file with custom metadata.
+```
+{
+  "metadata": {
+    "custom_features": {
+      "feature-level": {
+        "custom-feature-name": {
+          "dtype": "string",
+          "description": "(optional) description of the feature"
+        }
+      }
+    },
+    "custom_analyses": [
+      {
+        "cls_name": "BucketAnalysis",
+        "level": "feature-level",
+        "feature": "custom-feature-name",
+        "num_buckets": 15,
+        "method": "discrete/continuous"
+      }
+    ]
+  },
+  "examples": [
+    {
+      "output-json-key": "model-output",
+      "custom-feature-name": "feature-value"
+    }
+  ]
+}
+```
+where
+* `feature-level` represents the fine-grained level of the analysis
+    * `example` for sentence-level analysis
+    * `span` for span-level analysis
+* `dtype` represents the data type of the feature
+    * `float` for continuous feature
+    * `string` for discrete feature
+* `cls_name` means the analysis type
+    * `BucketAnalysis` requires the following fields to be defined
+    ```
+    BucketAnalysis(
```

Comment:
It might be a little bit confusing to suddenly have Python code here, as the expected input format is JSON. I'd suggest just having a bullet list. Similarly below.